### PR TITLE
Hiding channels for unavailable mentors

### LINKED ISSUE #119: Don't show mentor's channels if they not available
Must be done after #118 

Remove the channels from the mentors response for unavailable mentors (this will be reflected also in the UI). Similar to the request when the user is not loggedin.

https://github.com/Coding-Coach/find-a-mentor-api/blob/b88b00d36db7ba50a153cf82c1cafe313df32ebb/src/modules/mentors/mentors.controller.ts#L41

### PR DESCRIPTION
**Pull request by crysfel**
When a mentor is not available we want to hide the channels from users

closes #119

### CODE REVIEW

File: src/modules/common/mentors.service.ts [hunk 1]
```diff
@@ -22,7 +22,19 @@ export class MentorsService {
     const onlyMentors: any = {
       roles: 'Mentor',
     };
-    const projections = this.getMentorFields(isLoggedIn);
+    const projections = this.getMentorFields();
+
+    if (isLoggedIn) {
+      // Filter by availability
+      if (filters.available !== undefined) {
```
Comment (moshfeu) [inline src/modules/common/mentors.service.ts]:
```suggestion
      if (filters.available) {
        onlyMentors.available = filters.available;
        projections.channels = filters.available;
      }
        
```

Comment (crysfel) [reply src/modules/common/mentors.service.ts]: Tried that first, mongo complaints about mixing projections 🤷‍♂️

File: src/modules/common/mentors.service.ts [hunk 2]
```diff
@@ -134,8 +146,8 @@ export class MentorsService {
     return await this.applicationModel.deleteMany({ user }).exec();
   }
 
-  getMentorFields(isLoggedIn: boolean): any {
-    let projections: any = {
+  getMentorFields(): any {
+    const projections: any = {
```
Comment (moshfeu) [inline src/modules/common/mentors.service.ts]: It's not in that PR but why any?

File: src/modules/mentors/mentors.controller.ts [hunk 3]
```diff
@@ -39,20 +39,27 @@ export class MentorsController {
   @Get()
   @UsePipes(new PaginationPipe())
   async index(@Req() request: Request, @Query() filters: MentorFiltersDto) {
-    const data = await this.mentorsService.findAll(filters, !!request.user);
+    const available = await this.mentorsService.findAll({ ...filters, available: true }, !!request.user);
```
Comment (moshfeu) [inline src/modules/mentors/mentors.controller.ts]: That totally not my strong side but isn't better to call once to the database and do the filter (split to available and unavailable) in the code?

Comment (crysfel) [reply src/modules/mentors/mentors.controller.ts]: Given that we only have about 500 mentors and that later on we are planning to support pagination, this makes sense. 

I was originally thinking on avoiding looping though thousands of records, but that won't be case in the long term.

**Pull request merged**